KAFKA-17388: Remove broker-list in VerifiableProducer

LoganZhuZzz opened the pull request:
The broker-list option is no longer supported in version 4.0.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

LoganZhuZzz committed:
KAFKA-17388: Remove broker-list in VerifiableProducer
The broker-list option is no longer supported in version 4.0.

frankvicky reviewed (approved):
LGTM

LoganZhuZzz commented:
> LGTM

Thanks for your review.

chia7712 commented on tools/src/main/java/org/apache/kafka/tools/VerifiableProducer.java:
```diff
@@ -131,14 +131,6 @@ private static ArgumentParser argParser() {
                 .dest("bootstrapServer")
                 .help("REQUIRED: The server(s) to connect to. Comma-separated list of Kafka brokers in the form HOST1:PORT1,HOST2:PORT2,...");
 
-        connectionGroup.addArgument("--broker-list")
```
@LoganZhuZzz could you please update e2e also? see https://github.com/apache/kafka/blob/11966a209a8bb5bbf867b5bb5ca1d60b80e26650/tests/kafkatest/services/verifiable_producer.py#L227

noted that we should use `--bootstrap-server` only if the version >= 2.5. otherwise, we should keep using  `--broker-list`

LoganZhuZzz replied:
Thank you for the feedback! I appreciate the clarification regarding the --bootstrap-server flag. I'll update the code to ensure that --bootstrap-server is only used when the Kafka version is 2.5 or higher. For lower versions, I'll retain the use of --broker-list as suggested.

chia7712 reviewed (commented):
@LoganZhuZzz thanks for this patch. Please take a look at the major comment. It would be nice to fix e2e in this PR :)

LoganZhuZzz committed:
Fix: Use --bootstrap-server only for Kafka version 2.5 and higher

Responding to code review feedback, updated the code to use --bootstrap-server
flag only when the Kafka version is 2.5 or higher. For versions lower than 2.5,
retained the use of --broker-list as per the recommendation.

FrankYang0529 reviewed (approved):
LGTM. I can run `TC_PATHS="tests/kafkatest/sanity_checks/test_verifiable_producer.py" /bin/bash tests/docker/run_tests.sh` without error.

chia7712 reviewed (approved):
LGTM
